RPC Backend Registry

xush6528 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26919 RPC Backend Registry**

Make the names more systematic

Differential Revision: [D17262797](https://our.internmc.facebook.com/intern/diff/D17262797/)

xush6528 committed:
RPC Backend Registry

Make the name more systematic

Differential Revision: [D17262797](https://our.internmc.facebook.com/intern/diff/D17262797/)

[ghstack-poisoned]

xush6528 committed:
Update on "RPC Backend Registry"


Make the name more systematic

Differential Revision: [D17262797](https://our.internmc.facebook.com/intern/diff/D17262797/)

[ghstack-poisoned]

mrshenli commented on torch/distributed/rpc.py:
```diff
@@ -5,7 +5,7 @@
 from . import WorkerId
 from .internal_rpc_utils import serialize, PythonUDF
-from .rpc_backend_handler import is_backend_registered, registered_init_rpc
+from .rpc_init_handler_registry import is_backend_registered, rpc_init
```
Does it make sense if we call this `rpc_backend_registery` and `init_backend`? Because we already have a `init_rpc` API in `rpc.py`.

xush6528 replied:
Updating

mrshenli reviewed (approved):
LGTM! 

nit: this can be done in a followup PR. Shall we add some dummy backend implementation for testing? I believe right now we only have tests for invalid backend names.

xush6528 commented:
> LGTM!
> 
> nit: this can be done in a followup PR. Shall we add some dummy backend implementation for testing? I believe right now we only have tests for invalid backend names.

Agree. Each component should have corresponding tests. 

We shouldn't make them get coverage brought by upper-level code.

xush6528 committed:
Update on "RPC Backend Registry"


Make the name more systematic

Differential Revision: [D17262797](https://our.internmc.facebook.com/intern/diff/D17262797/)

[ghstack-poisoned]

pietern commented on torch/distributed/rpc.py:
```diff
@@ -5,7 +5,7 @@
 from . import WorkerId
 from .internal_rpc_utils import serialize, PythonUDF
-from .rpc_backend_handler import is_backend_registered, registered_init_rpc
+from .rpc_backend_registry import is_rpc_backend_registered, init_rpc_backend
```
If I'm not mistaken, these functions will now be available on the `torch.distributed.rpc` module.

Can you add an `__all__` to make sure only the functions we want are available on the module?

pietern replied:
@xush6528 Can you ack this comment?

xush6528 replied:
@pietern 

Sorry.

Will it be better if we don't use `from "something" import *`?

pietern replied:
Thanks.

I agree, but that puts the burden on the user. I think it's good practice to define the interface and define what should be exported and what not. This is not done everywhere consistently, but still there are plenty hits for both `__all__` and `import *` in the entire code base.

xush6528 replied:
I think rpc should create a folder like `distributed/autograd` does.
Maintain the names want to expose to users in `torch/distributed/autograd/__init__.py`.
Do the actual implmentation in `torch/distributed/rpc/rpc_api.py`.
